### ISSUE TITLE
Impenetrable Shadow Zones

### ISSUE THREAD

**Issue by Airova** [anegostudios/VintageStory-Issues#353]
**Game Version:** 1.11.6
**Platform:** Windows 8.1 64-bit
**Modded:** Yes (CarryCapacity, Salt Production, a few manual tweaks to fuel production and duration)

### Description
Some areas in the world are permanently in shadow no matter what time of day it is. Light sources do work, so if I walk through carrying a torch I can see around me, but aside from that it's pitch blackness. I've seen this in a previous version of the game as well, but it was quite a while ago, don't remember which. We had this issue on the Simpletons server back when that was active. I found several of these patches, all near each other (so far) but not all connected.

### How to reproduce
Not sure what the conditions are. I wandered around the map exploring and just happened upon some of these zones. They appear to fall along chunk borders and have straight lines as edges.

### Screenshots
![20191209224226_1](https://user-images.githubusercontent.com/13468600/70503373-89ec6e00-1b23-11ea-9ccf-2ce8e85b6d5d.jpg)
![20191209224215_1](https://user-images.githubusercontent.com/13468600/70503380-8d7ff500-1b23-11ea-999b-aa8d32363b81.jpg)
![20191209224313_1](https://user-images.githubusercontent.com/13468600/70503391-91137c00-1b23-11ea-9fad-609df495d42b.jpg)
![20191209224317_1](https://user-images.githubusercontent.com/13468600/70503398-940e6c80-1b23-11ea-8cc2-638efb7a3787.jpg)
(In the last screen shot you can see both the zone on the black sand and, in the background, top left of the image, the zone on the grassy plain below, showing their relative position.)

**Comment by tyronx** [comment]
I've seen those happening whenever a singleplayer or multiplayer server crashed. It seems in those cases the chunks don't receive their final lighting pass. Don't know the cause yet, but you can fix them manually chunk-by-chunk by standing inside them and typing `/debug cr`

**Comment by Airova** [comment]
Will do that, thanks for the short-term fix! 👍

**Comment by tyronx** [comment]
Ohh, Aira, could you enter one of those checks and type `/wgen pos ymax` and tell me what value it prints please.

**Comment by Airova** [comment]
I am at work right now but I will check it in the evening when I'm home.

**Comment by Airova** [comment]
I tried a few different chunks and got several different values:
121
130
119
150
148
119

**Comment by Doug5000** [comment]
Hello, recently had this issue in my single player, unmodded world on a fresh install. I have tried using debug tools to fix them but to no effect. I do no recall having a crash at any point, but I did have some sort of message explaining that the world needed to be regenerated because of an update. Pretty sure I noticed the dark areas afterwards.

 Gameplay wise, is my best option to simply generate another world and hope for the best?

**Comment by radfast** [comment]
Please see my [comment today](https://github.com/anegostudios/VintageStory-Issues/issues/3253#issuecomment-2558712098) on issue #3253.

TL;DR:
1. The dev team believes the most common causes of dark chunks in savegames have been resolved in game versions 1.18.0 and 1.18.2
2. Dark chunks can still occur rarely, for unknown reasons but most likely to be previous crashes / exceptions during the worldgen of the affected chunk.  It is quite likely that those previous issues were in a previous run of the server or a previous load of the affected savegame, some hours, days or weeks previously.  (Essentially it could be any issue which interrupted things before the chunk was saved following the application of sunlight to it, as the final stage in worldgen. For the devteam, more precisely for some reason the chunk lighting has not been saved but the MapChunk _has_ been saved therefore worldgen is flagged as done.)
3. In some cases the underlying issue may have been permanently logged, with the name of the save game and the chunk coordinates, in the log file `server-worldgen.log` - **so affected players or servers should check that logfile please**.
4. **In all cases, the issue can be fixed by a player going to the affected chunk, standing inside it, and using command `/debug chunk relight`**.

@noelle-lavenza In light (haha) of the above while acknowledging that dark chunks issues can still occur and we are still keen to find the root cause or a general/automated way to resolve this, I don't think its helpful to keep a general issue about this open dating from game version 1.11 and year 2019.   Affected players probably will not even discover this issue, as it's so old.  I think it seems preferable for affected players in game versions 1.20+ to open a new issue now, including relevant recent logs.
